app/controllers/concerns/course/forum/topics_hide_concern.rb

```diff
@@ -0,0 +1,29 @@
+module Course::Forum::TopicsHideConcern
+  extend ActiveSupport::Concern
+
+  def set_hidden
+    authorize! :set_hidden, @topic
+
+    if @topic.update_attributes(hidden_params)
+      redirect_to course_forum_topic_path(@course, @forum, @topic),
+                  success: t('course.forum.topics.hidden.success', state: hidden_state)
+    else
+      redirect_to course_forum_topic_path(@course, @forum, @topic),
+                  danger: t('course.forum.topics.hidden.failure', state: hidden_state)
+    end
+  end
+
+  private
+
+  def hidden_state
```

Comment:
`hidden_state_text` because otherwise it sounds like a property getter of the controller